recommenders/evaluation/spark_evaluation.py

```diff
@@ -158,7 +159,8 @@ def exp_var(self):
             0
         ]
         var2 = self.y_pred_true.selectExpr("variance(label)").collect()[0][0]
-        return 1 - var1 / var2
+        # numpy divide is more tolerant to var2 being zero
+        return 1 - np.divide(var1, var2)
```

Comment:
I'm looking at the test here: https://github.com/microsoft/recommenders/blob/main/tests/unit/recommenders/evaluation/test_spark_evaluation.py#L239, should we maybe add the edge case for when var2 is zero?

Reply:
Maybe, although it is a very unusual case in practice (everything in the data is rated equally).